src/Processors/Formats/Impl/ProtobufRowInputFormat.h

```diff
@@ -61,12 +69,36 @@ class ProtobufSchemaReader : public IExternalSchemaReader
 };
 
 /// proton: starts
+/// Confluent framing + Protobuf binary datum encoding. Mainly used for Kafka.
+/// Uses 3 caches:
+/// 1. global: schema registry cache (base_url -> SchemaRegistry)
+/// 2. SchemaRegistry: schema cache (schema_id -> schema)
+/// 3. ProtobufConfluentRowInputFormat: deserializer cache (schema_id -> AvroDeserializer)
+/// This is needed because KafkaStorage creates a new instance of InputFormat per a batch of messages
```

Comment:
Wondering how do we do cache invalidation ? do we support it now and how it works in prod ?